nextcord/application_command.py

```diff
@@ -0,0 +1,883 @@
+            raise InvalidCommandType(f"{self.type} is not a handled Application Command type.")
+
+    async def call_invoke_slash(self, state: ConnectionState, interaction: Interaction,
+                                option_data: List[Dict[str, Any]]):
+        """|coro|
+
+        This invokes the slash command implementation with the given raw option data to turn into proper kwargs for
+        the callback.
+
+        Parameters
+        ----------
+        state: :class:`ConnectionState`
+            State object used to fetch Guild, Channel, etc from cache.
+        interaction: :class:`Interaction`
+            Interaction object to pass to the callback.
+        option_data: :class:`list`
+            List of option data dictionaries from interaction data payload.
+        """
+        kwargs = {}
+        uncalled_args = self.arguments.copy()
+        for arg_data in option_data:
+            if arg_data["name"] in uncalled_args:
+                uncalled_args.pop(arg_data["name"])
+                kwargs[self.arguments[arg_data["name"]].functional_name] = \
+                    self.arguments[arg_data["name"]].handle_slash_argument(state, arg_data["value"], interaction)
+            else:
+                # TODO: Handle this better.
+                raise NotImplementedError(f"An argument was provided that wasn't already in the function, did you"
+                                          f"recently change it?\nRegistered Args: {self.arguments}, Discord-sent"
+                                          f"args: {interaction.data['options']}, broke on {arg_data}")
+        for uncalled_arg in uncalled_args.values():
+            kwargs[uncalled_arg.functional_name] = uncalled_arg.default
+        await self.invoke_slash(interaction, **kwargs)
+
+    async def invoke_slash(self, interaction: Interaction, **kwargs):
+        """|coro|
+
+        Invokes the callback with the kwargs given."""
+        if self.cog_parent:
+            await self.callback(self.cog_parent, interaction, **kwargs)
+        else:
+            await self.callback(interaction, **kwargs)
+
+    def error(self, coro):
+        # TODO: Parity with legacy commands.
+        raise NotImplementedError
+
+    @property
+    def payload(self) -> dict:
+        """:class:`dict`: Outputs a Discord "Application Command Option Structure" payload."""
+        self._analyze_content()
+        ret = {"type": self.type.value, "name": self.name, "description": self.description}
+        if self.choices:
+            ret["choices"] = [{key: value} for key, value in self.choices.items()]
+        if self.children:
+            ret["options"] = [child.payload for child in self.children.values()]
+        elif self.arguments:
+            ret["options"] = [argument.payload for argument in self.arguments.values()]
+        return ret
+
+    def subcommand(self, **kwargs):
+        """Creates a new subcommand off of this one."""
```

Comment:
This function appears public, care to document it correct or point to document explaining the ``**kwargs`` it takes

Reply:
yeah maybe link to the ``ApplicationSubcommand`` docs to show what kwargs is